logstash-core/lib/logstash/shutdown_controller.rb

```diff
@@ -47,10 +47,10 @@ def start
       cycle_number = 0
       stalled_count = 0
       Stud.interval(@cycle_period) do
-        @reports << Report.from_pipeline(@pipeline)
+        @reports << pipeline_report_snapshot
         @reports.delete_at(0) if @reports.size > @report_every # expire old report
         if cycle_number == (@report_every - 1) # it's report time!
-          logger.warn(@reports.last.to_hash)
+          logger.warn(@reports.last.to_simple_hash)
```

Comment:
You don't really need to explicitly call `to_simple_hash` here, you can implement `to_str` in the snapshot class, when the string conversion occur ruby will call `to_str`